### ISSUE TITLE
unexpected behaviour with --set-default-module --robot

### ISSUE THREAD

**Issue by lucamar** [easybuilders/easybuild-framework#3090]
The EasyBuild option `--set-default-module` will apply to all the recipes in the same command line: this means that every dependency of the build using `--set-default-module` will be set default, overwriting existing  defaults for different versions of the same module name. I paste here the info on the flag:
```
eb -H | grep default
    --set-default-module
                        Set the generated module as default (def False)
```
This is happening frequently with `CMake`, since there are builds using different `CMake` versions.

I provide an example below with Bison, that depends on M4 (the lower case naming scheme is enabled on the build system and I skip lines of output using the symbol `[...]` to make it shorter): 
```
eb Bison-3.3.2.eb -Dr
[...]
 * [ ] $CFGS/m/M4/M4-1.4.18.eb (module: m4/1.4.18)
 * [ ] $CFGS/b/Bison/Bison-3.3.2.eb (module: bison/3.3.2)

eb Bison-3.3.2.eb -r --set-default-module
 [...]

module avail
------------------------------------------------------------------ 
bison/3.3.2(default)  m4/1.4.18(default)
```

**Comment by boegel** [comment]
Thanks for the bug report @lucamar!

It's certainly not the intention that `--set-default-module` applies to all dependencies, it should indeed only apply to the modules that correspond with the easyconfig files explicitly listed.

I'll try and get this fixed ASAP...

**Comment by boegel** [comment]
Fix implemented in da2978d, I'll open the PR to include the fix in EasyBuild v4.1.0 as soon as #3093 gets merged...

**Comment by lucamar** [comment]
Thanks a lot @boegel!
